src/main/java/org/opensearch/flowframework/workflow/GetResourceStep.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.support.PlainActionFuture;
+import org.opensearch.flowframework.model.ResourceCreated;
+
+import java.util.Map;
+
+/**
+ * Internal step to pass created resources to dependent nodes. Only used in reprovisioning
+ */
+public class GetResourceStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(GetResourceStep.class);
+    private final ResourceCreated resourceCreated;
+
+    /** The name of this step */
+    public static final String NAME = "get_resource";
```

Comment:
Would prefer a more verbose name. Since the state index has `resources_created` then adding the word `created` to the name would help align those fields, since this is essentially a "getter" for that.   Alternately `get_state_resource` might work.

Reply:
Ill change this to `get_state_resource`